### PR TITLE
Correct the wrong format string in vehicle.cpp

### PR DESCRIPTION
**Pull request by ghost**
The mixed use of printf modifiers with argument selection `%1$s` and without argument selection `%s` in `_vscprintf_p` function (provided by MSVC) causes crash. Can be proved with the following test program.

``` C
#include <stdio.h>
#include <stdarg.h>

void fmt(const char* format, ...)
{
    va_list ap;
    va_start(ap, format);
    _vscprintf_p(format, ap);
    va_end(ap);
}

int main()
{
    fmt("%1$s and %s and %2$s\n", "arg1", "arg2", "arg3");
    return 0;
}
```

It crashes.

The problem has remained undiscovered is because the implementation of `vstring_format` function in `output.cpp` is different for different compilers, Microsoft's `_vscprintf_p` function is invoked if the compiler is MSVC(`output.cpp:1668`), and it seems `_vscprintf_p` is more picky.

### CODE REVIEW

**Comment by DLaboratory** [comment]
BTW, I met this bug this morning, Cataclysm crashed. How can I rescue my save from memory dump? I farmed and crafted a lot of things, but the program crashed, my work is all lost. :cry:

**Comment by mugling** [comment]
It might be possible to reanimate the coredump and extract `player` but I'd probably just use the debug menu

File: src/vehicle.cpp [hunk 1]
```diff
@@ -4838,7 +4838,7 @@ void vehicle::handle_trap( const tripoint &p, int part )
     }
     if( g->u.sees(p) ) {
         if( g->u.knows_trap( p ) ) {
-            add_msg(m_bad, _("The %1$s's %s runs over %2$s."), name.c_str(),
```
Comment (mugling) [inline src/vehicle.cpp]: Please provide also a translation summary above this line

Comment (DLaboratory) [inline src/vehicle.cpp]: How?

Comment (mugling) [inline src/vehicle.cpp]: via `//~` syntax then document the parameters

**Pull request merged**